File: ZigZag Conversion.java
public class Solution {
    public String convert(String s, int nRows) {
        if (nRows == 1)
            return s;
        String[] arr = new String[nRows];
        for (int i = 0; i < nRows; i++) {
            arr[i] = new String();
        }
        int r = 0;
        boolean down = true;
        for (char ch : s.toCharArray()) {
            arr[r] += ch;
            if (down) {
                r++;
                if (r >= nRows - 1)
                    down = false;
            }
            else {
                r--;
                if (r <= 0) 
                    down = true;
            }
        }
        String ret = "";
        for (String str : arr) {
            ret += str;
        }
        return ret;
    }
}

File: ZigZag Conversion.py
class Solution:
	# @return a string
	def convert(self, s, nRows):
		if (nRows == 1):
			return s
		strs = [""] * nRows
		r = 0
		down = True
		for ch in s:
			strs[r] += ch
			if (down):
				r = r + 1
				if (r == nRows - 1):
					down = False
			else:
				r = r - 1
				if (r == 0):
					down = True
		ret = ""
		for str in strs:
			ret = ret + str
		return ret